lib/filterx/filterx-scope.c

```diff
@@ -91,6 +96,50 @@ filterx_scope_new(void)
   return self;
 }
 
+static FilterXScope *
+filterx_scope_clone(FilterXScope *other)
+{
+  FilterXScope *self = filterx_scope_new();
+
+  GHashTableIter iter;
+  gpointer _key, _value;
+
+  g_hash_table_iter_init(&iter, self->value_cache);
+  while (g_hash_table_iter_next(&iter, &_key, &_value))
+    {
+      NVHandle handle = GPOINTER_TO_INT(_key);
+      FilterXObject *value = (FilterXObject *) _value;
```

Comment:
Shouldn't we have this here as well?
```
value->shadow = TRUE;
```

Maybe we should use `filterx_scope_register_message_ref()` explicitly, like:
```c
      FilterXObject *cloned_value = filterx_object_clone(value);
      filterx_scope_register_message_ref(self, handle, cloned_value);
      filterx_object_unref(cloned_value);
```

Reply:
My latest iteration changed a lot here, so I think this is now outdated. Closing, but let me know if you disagree.